cardano-node/src/Cardano/Node/Run.hs

```diff
@@ -325,31 +357,60 @@ handleSimpleNode p trace nodeTracers npm onKernel = do
 
          when mockValidateDB $ traceWith tracer "Performing DB validation"
 
+  stopOnSlotSynced :: NodeProtocolMode -> MaxSlotNo
+  stopOnSlotSynced (RealProtocolMode cli) = shutdownOnSlotSynced cli
+  stopOnSlotSynced _ = NoMaxSlotNo
+
+  -- | Produce a potential pair of FDs:
+  --  - read end, where EOF is a signal of termination
+  --  - write end, which is means of internal termination handling
+  decideShutdownFds :: NodeProtocolMode -> IO ShutdownFDs
+  decideShutdownFds (RealProtocolMode NodeCLI{shutdownIPC = Just fd}) = pure $ ExternalShutdown fd
+  decideShutdownFds (RealProtocolMode NodeCLI{shutdownOnSlotSynced = MaxSlotNo{}}) = do
+    (r, w) <- Posix.createPipe
```

Comment:
I _think_ windows must have that.

Hmm, https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/pipe?view=vs-2019 is close, but https://hackage.haskell.org/package/unix-2.7.2.2/docs/src/System.Posix.IO.Common.html#createPipe doesn't use that.

Reply:
Aha, found a portable option!

http://hackage.haskell.org/package/process-1.6.8.2/docs/System-Process.html#v:createPipe